music_assistant/controllers/player_queues.py

```diff
@@ -1129,22 +1199,35 @@ async def _load_item(
             )
         )
         current_index = self.index_by_id(queue_id, queue_item.queue_item_id)
-        previous_track_from_same_album = (
-            (previous_index := max(current_index - 1, 0))
-            and (previous_index > 0)
-            and (previous_item := self.get_item(queue_id, previous_index))
-            and (
-                queue_item.media_item
-                and hasattr(queue_item.media_item, "album")
-                and queue_item.media_item.album
-                and previous_item.media_item
+        if current_index is None:
+            previous_track_from_same_album = False
+        else:
+            previous_index = max(current_index - 1, 0)
+            previous_track_from_same_album = (
+                previous_index > 0
+                and (previous_item := self.get_item(queue_id, previous_index)) is not None
+                and previous_item.media_item is not None
                 and hasattr(previous_item.media_item, "album")
-                and previous_item.media_item.album
+                and previous_item.media_item.album is not None
+                and queue_item.media_item is not None
+                and hasattr(queue_item.media_item, "album")
+                and queue_item.media_item.album is not None
                 and queue_item.media_item.album.item_id == previous_item.media_item.album.item_id
             )
-        )
         playing_album_tracks = next_track_from_same_album or previous_track_from_same_album
         if queue_item.media_item and queue_item.media_item.media_type == MediaType.TRACK:
+            # Type narrowing with proper error handling
+            if not isinstance(queue_item.media_item, Track):
```

Comment:
This type narrowing does not make sense to me. If mypy complains, we could simply replace the if:
`if queue_item.media_item and queue_item.media_item.media_type == MediaType.TRACK:`
to
`if queue_item.media_item and isinstance(queue_item.media_item, Track):`

Reply:
Yep simple is better